dataeng/jobs/analytics/DBTAutomated.groovy

```diff
@@ -0,0 +1,68 @@
+package analytics
+import static org.edx.jenkins.dsl.AnalyticsConstants.secure_scm
+import static org.edx.jenkins.dsl.AnalyticsConstants.common_authorization
+import static org.edx.jenkins.dsl.AnalyticsConstants.common_log_rotator
+import static org.edx.jenkins.dsl.AnalyticsConstants.common_wrappers
+import static org.edx.jenkins.dsl.AnalyticsConstants.common_publishers
+import static org.edx.jenkins.dsl.AnalyticsConstants.common_triggers
+import static org.edx.jenkins.dsl.AnalyticsConstants.secure_scm_parameters
+
+class DBTAutomated{
+    public static def job = { dslFactory, allVars ->
+        dslFactory.job("dbt-automated"){
+            description(
+                "Automatically run dbt <strong>in production</strong>, overwriting data in the PROD database when Schema Builder generated PR are merged"
+            )
+            authorization common_authorization(allVars)
+            logRotator common_log_rotator(allVars)
+            parameters secure_scm_parameters(allVars)
+            parameters {
+                stringParam('NOTIFY', env_config.get('NOTIFY', allVars.get('NOTIFY','$PAGER_NOTIFY')), 'Space separated list of emails to send notifications to.')
+            }
+            environmentVariables {
+                env('WAREHOUSE_TRANSFORMS_URL', allVars.get('WAREHOUSE_TRANSFORMS_URL'))
+                env('WAREHOUSE_TRANSFORMS_BRANCH', allVars.get('WAREHOUSE_TRANSFORMS_BRANCH'))
+                env('DBT_TARGET', allVars.get('DBT_TARGET'))
+                env('DBT_PROFILE', allVars.get('DBT_PROFILE'))
+                env('DBT_PROJECT_PATH', allVars.get('DBT_PROJECT_PATH'))
+                env('DBT_RUN_OPTIONS', allVars.get('DBT_RUN_OPTIONS'))
+                env('DBT_RUN_EXCLUDE', allVars.get('DBT_RUN_EXCLUDE'))
+            }
+            multiscm secure_scm(allVars) << {
+                git {
+                    remote {
+                        url('$WAREHOUSE_TRANSFORMS_URL')
+                        branch('$WAREHOUSE_TRANSFORMS_BRANCH')
+                    }
+                    extensions {
+                        relativeTargetDirectory('warehouse-transforms')
+                        pruneBranches()
+                        cleanAfterCheckout()
+                    }
+                }
+            }
+            triggers {
+                scm('H/2 * * * *')
```

Comment:
I don't think we should run this job this frequently. In fact, doing so would result in 99% of the build history being `It is not a Schema Builder commit. Exiting from job`, and it would be hard to track down which run actually introduced the change.

Ideally, we could use GH webhooks to trigger these with a push model, rather than needing to constantly pull. Perhaps we should revisit that decision at some point.

Reply:
Actually GH polling isn't that bad, It polls in the background and triggers only when it detects new commits so polling with a frequency of 2 minute is fine.

Reply:
But I would like to have a discovery ticket for GH webhooks may be it can be possible to apply